### PR TITLE
Only retrieve modification count instead of whole data

### PR DESCRIPTION
**Pull request by Meklo**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java [hunk 1]
```diff
@@ -562,6 +562,11 @@ public String getNetworkModifications(@NonNull UUID nodeUuid) {
         return getNetworkModifications(nodeUuid, false, true);
     }
 
+    @Transactional(readOnly = true)
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java:565]: Remove unsed method `getNetworkModifications` 
Please see all warnings before a push !

File: src/main/java/org/gridsuite/study/server/service/NetworkModificationService.java [hunk 2]
```diff
@@ -96,6 +97,20 @@ public String getModifications(UUID groupUUid, boolean stashedModifications, boo
         }
     }
 
+    public Integer getModificationsCount(UUID groupUUid, boolean stashedModifications) {
+        Objects.requireNonNull(groupUUid);
+        var path = UriComponentsBuilder.fromPath(GROUP_PATH + DELIMITER + NETWORK_MODIFICATIONS_COUNT_PATH)
+            .queryParam(QUERY_PARAM_ONLY_STASHED, stashedModifications)
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/NetworkModificationService.java:103]: Cahnge by `QUERY_PARAM_STASHED`

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 3]
```diff
@@ -1407,7 +1406,7 @@ public void moveStudyNode(UUID studyUuid, UUID nodeToMoveUuid, UUID referenceNod
         List<NodeEntity> oldChildren = null;
         checkStudyContainsNode(studyUuid, nodeToMoveUuid);
         checkStudyContainsNode(studyUuid, referenceNodeUuid);
-        boolean shouldInvalidateChildren = !EMPTY_ARRAY.equals(networkModificationTreeService.getNetworkModifications(nodeToMoveUuid));
+        boolean shouldInvalidateChildren = networkModificationTreeService.getNetworkModificationsCount(nodeToMoveUuid, false) > 0;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:1409]: Add a method in the `networkModificationTreeService:hasModifications(noduid, stahed)` ?

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 4]
```diff
@@ -1396,7 +1395,7 @@ public void duplicateStudyNode(UUID sourceStudyUuid, UUID targetStudyUuid, UUID
         checkStudyContainsNode(sourceStudyUuid, nodeToCopyUuid);
         checkStudyContainsNode(targetStudyUuid, referenceNodeUuid);
         UUID duplicatedNodeUuid = networkModificationTreeService.duplicateStudyNode(nodeToCopyUuid, referenceNodeUuid, insertMode);
-        boolean invalidateBuild = !EMPTY_ARRAY.equals(networkModificationTreeService.getNetworkModifications(nodeToCopyUuid));
+        boolean invalidateBuild = networkModificationTreeService.getNetworkModificationsCount(nodeToCopyUuid, false) > 0;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:1398]: See comment below

File: src/main/java/org/gridsuite/study/server/service/StudyService.java (hunk 3 above)
Comment (Meklo) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java]: Done

File: src/main/java/org/gridsuite/study/server/service/StudyService.java (hunk 4 above)
Comment (Meklo) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java]: Done in all occurences

File: src/main/java/org/gridsuite/study/server/service/NetworkModificationService.java (hunk 2 above)
Comment (Meklo) [reply src/main/java/org/gridsuite/study/server/service/NetworkModificationService.java]: Done

File: src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java (hunk 1 above)
Comment (Meklo) [reply src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java]: Done

**Review by souissimai — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**